bpdm-pool/src/main/kotlin/org/eclipse/tractusx/bpdm/pool/service/RequestValidationService.kt

```diff
@@ -272,117 +293,111 @@ class RequestValidationService(
     ): Map<AddressPartnerUpdateRequest, Collection<ErrorInfo<AddressUpdateError>>> {
 
         val addressRequests = requests.map { it.address }
-        val addressMetadata = metadataService.getMetadata(addressRequests).toKeys()
-
         val requestedAddressBpns = requests.map { it.bpna }
         val existingAddressBpns = addressRepository.findDistinctByBpnIn(requestedAddressBpns).map { it.bpn }.toSet()
 
-        val addressDuplicateIdentifierCandidates = getAddressDuplicateIdentifierCandidates(addressRequests)
+        val regionValidator = ValidateRegionExists(addressRequests, AddressUpdateError.RegionNotFound)
+        val identifiersValidator = ValidateIdentifierTypesExists(addressRequests, AddressUpdateError.IdentifierNotFound)
+        val identifiersDuplicateValidator = ValidateAddressIdentifiersDuplicated(addressRequests, AddressUpdateError.AddressDuplicateIdentifier)
+        val existingBpnValidator = ValidateUpdateBpnExists(existingAddressBpns, AddressUpdateError.AddressNotFound)
 
         return requests.flatMap { request ->
             val address = request.address
 
-            val validationErrors = validateRegionExists(address, addressMetadata.regions, AddressUpdateError.RegionNotFound, request) +
-                    validateIdentifierTypesExists(address, addressMetadata.idTypes, AddressUpdateError.IdentifierNotFound, request) +
-                    validateUpdateBpnExists(request.bpna, existingAddressBpns, AddressUpdateError.AddressNotFound) +
-                    validateAddressIdentifiersDuplicated(
-                        address = address,
-                        existingIdentifiers = addressDuplicateIdentifierCandidates,
-                        bpn = request.bpna,
-                        error = AddressUpdateError.AddressDuplicateIdentifier,
-                        entityKey = request
-                    )
-
+            val validationErrors = regionValidator.validate(address, request) +
+                    identifiersValidator.validate(address, request) +
+                    identifiersDuplicateValidator.validate(address, request, request.bpna) +
+                    existingBpnValidator.validate(request.bpna)
             validationErrors.map { Pair(request, it) }
 
         }.groupBy({ it.first }, { it.second })
     }
 
-
-    private fun <ERROR : ErrorCode> validateIdentifierTypesExists(
-        request: IBaseLegalEntityDto,
-        existingTypes: Set<String>,
-        error: ERROR,
-        entityKey: RequestWithKey
-    )
-            : Collection<ErrorInfo<ERROR>> {
-        val requestedTypes = request.identifiers.map { it.type }
-        val missingTypes = requestedTypes - existingTypes
-
-        return missingTypes.map {
-            ErrorInfo(
-                error,
-                "Legal Entity Identifier Type '$it' does not exist",
-                entityKey.getRequestKey()
-            )
+    inner class ValidateIdentifierLeTypesExists<ERROR : ErrorCode>(
```

Comment:
Maybe call them `Validator...` not `Validate...`

Reply:
In this step the class is called like the refactored methods to make the refactoring clearer.
Renaming could be done in the next step